pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -7702,8 +7702,11 @@ func TestChangefeedKafkaMessageTooLarge(t *testing.T) {
 
 			rnd, _ := randutil.NewPseudoRand()
 
+			maxFailures := 200
+			numFailures := 0
```

Comment:
Does this need to use atomics?

Reply:
Yes it does

Reply:
The sync producer and async producer are two separate goroutines that use it. I don't see much synchronization between them.